src/parsers.rs

```diff
@@ -288,13 +297,30 @@ fn duration_minute(i: &[u8]) -> IResult<&[u8], u32> {
 // S[S][[,.][MS]]
 fn duration_second_and_millisecond(i: &[u8]) -> IResult<&[u8], (u32, u32)> {
     let (i, s) = m_to_n_digit_in_range(i, 1, 2, 0..=60)?;
-    let (i, ms) = opt(map(preceded(one_of(",."), fractions), duration_millisecond))(i)?;
+    let (i, ms) = opt(preceded(one_of(",."), duration_millisecond))(i)?;
 
     Ok((i, (s, ms.unwrap_or(0))))
 }
 
-fn duration_millisecond(fraction: f32) -> u32 {
-    (1000.0 * fraction) as u32
+fn duration_millisecond(i: &[u8]) -> IResult<&[u8], u32> {
```

Comment:
Hm, `time_millisecond` and `duration_millisecond` look exactly similar.
We can probably get rid of one of these now (but I can also do that in a followup)

Reply:
Happy to do that if that is what you want, I was just trying to minimize structural changes.

Reply:
Yeah, I wouldn't mind if you refactor that. Best done in a separate commit.